### DISCUSSION TITLE
Smart meter WIFI gateway with amsreder on a Kamstrup with HAN-P1 module no HAN data.

### DISCUSSION THREAD

**Discussion by jorgenlidholm** [UtilitechAS/amsreader-firmware#589] [Q&A] [answered]
Hi,
I bought a Slimme meter wifi gateway from smartgateways.nl and did not get it to run on my Kamstrup with HAN P1 module (RJ-12).
A friend told me he gave up after some discussions with smartgateways support. He was told they where not compatible.

I found this project and thought I should give it a go.
I managed to flash my device using esp32 firmware and everything except for the HAN data works nicely.

Does any one know if there is any chance it will work for me? I have currently the DSMR / EFS2 / HAN P1 profile activated on the power meter.
Best regards,
 Jörgen

**Comment by gskjold** [comment]
I don't know what GPIO the slimme meter wifi gateway is using. I would suggest trying setting HAN GPIO to UART2 and baud config to 115200 8N1. The data payload should be supported by this firmware. If you have any issues, you could try to enable debug mode with telnet and set it to verbose. Telnet to the device, and you should see the raw data there. https://amsleser.no/blog/post/24-telnet-debug

Note that using the HAN port requires that the grid company have enabled the port for you. There is also the question of encryption, but only your grid company can answer that.

**Reply by jorgenlidholm** [reply]
Thank you for responding!

The HAN port is activated by the grid company, with the HAN P1 profile. And there should be no encryption.
I've yet to find a configuration that matches the serial config of the HAN module. I guess I should try them all.

**Comment by ArnieO** [comment]
> Hi, I bought a Slimme meter wifi gateway from smartgateways.nl and did not get it to run on my Kamstrup with HAN P1 module (RJ-12). A friend told me he gave up after some discussions with smartgateways support. He was told they where not compatible.
> 
> I found this project and thought I should give it a go. I managed to flash my device using esp32 firmware and everything except for the HAN data works nicely.
> 
> Does any one know if there is any chance it will work for me? I have currently the DSMR / EFS2 / HAN P1 profile activated on the power meter. Best regards, Jörgen

You will at least have to power the Slimmelezer from USB, as the Kamstrup HAN P1 module does not deliver 5V to the reader. After that I think it boils down to finding our which GPIO the Slimmelezer uses, and use that in the setting of our firmware.

**Reply by jorgenlidholm** [reply]
Thank you for responding!
I'm powering my Slimmelezer using the USB-port. Maybe I can open the casing and investigate more closely which pin is connected.

**Comment by jorgenlidholm** [comment, marked as answer]
So I cracked the case open and could find out that the Data pin is mapped to GPIO18 and that the data request/Data Out is indeed +5V.
Then I managed to find the correct serial configuration as well, 2400 8n1 Inverted.
The Kamstrup Omnipower has the HAN P1 profile activated and the meter is equipped with the HAN P1 Module (RJ12).

**Reply by gskjold** [reply]
Nice! Is the chip maybe ESP32-S2 ? GPIO18 is UART1 on that chip.

**Reply by ArnieO** [reply]
You cannot use the Slimmelezer (or our product Pow-P1) on the Kamstrup meter with HAN P1 module, as the Kamstrup does not deliver 5V on the RJ12 plug of the adapter. As you will see this document you need to provide 5V yourself: https://documentation.kamstrup.com/docs/HAN_P1/en-GB/Information_sheet/CONTAAA62C8B611D4C4EB3186FD81AEE0152/
(This is a problem only with Kamstrup; it does not follow the P1 standard.)

The bitrate is 115 200 baud on a port configured as P1, not 2400. 
8N1 is correct parity. 

Our current generation Pow-K+ needs a small modification to be used on P1-configured meters. This will be taken care of in next version that will be in sale when current stock is depleted - which should be somewhere this autumn. It will still take its operating power from the meter, no external power needed.

**Reply by ArnieO** [reply]
I believe Slimmelezer is still ESP8266?

**Reply by jorgenlidholm** [reply]
It is not the Slimmelezer. Some where some how I got the name Slimme meter, I guess it's the Dutch name. My mistake!
I've updated the title of this discussion.
This is the device: https://smartgateways.nl/en/product/smart-meter-wifi-gateway/

It does not work with UART1, I think it's a ESP-WROOM-32 with an extension board for the RJ12 socket.

**Reply by ArnieO** [reply]
OK, interesting. There are others running our firmware on the Slimmelezer, but I do not know if anyone has tried it on this one.
You need to note the type of ESP32 module to be sure you flash the right firmware, and you need to be able to set it in programming mode to flash the firmware. If it is an (old) ESP32-WROOM it cannot be programmed over USB unless the board has USB-to-serial circuitry. 
For my curiosity: Could you take photos of the board and post here?

**Reply by jorgenlidholm** [reply]
I did flash it using esptool.py with --chip esp32.
Holding the reset button while executing espflash until recognized works.

The photos are not great, but should be clear enough I think.
![bild](https://github.com/UtilitechAS/amsreader-firmware/assets/1193206/a5167963-d788-4857-9721-877214e6511a)
![bild](https://github.com/UtilitechAS/amsreader-firmware/assets/1193206/6a0c0d7d-48c0-4a16-8cfd-90105784eaa6)
![bild](https://github.com/UtilitechAS/amsreader-firmware/assets/1193206/237c407d-6f84-4a82-86f6-196b14d0a0da)

**Reply by ArnieO** [reply]
Thank you; interesting! 
Apparently some generic development board with a quite simple interface module added. Flashing the ESP is no problem then.
